ipa-core/src/protocol/hybrid/oprf.rs

```diff
@@ -0,0 +1,262 @@
+                value: value.clone(),
+                breakdown_key: breakdown_key.clone(),
+            }
+        })
+        .collect())
+}
+
+#[cfg(test)]
+mod test {
+
+    use crate::{
+        ff::boolean_array::{BA3, BA8},
+        protocol::hybrid::oprf::compute_prf_for_inputs,
+        report::hybrid::{HybridReport, IndistinguishableHybridReport},
+        secret_sharing::IntoShares,
+        test_fixture::{
+            flatten3v, hybrid::TestHybridRecord, RoundRobinInputDistribution, TestWorld,
+            TestWorldConfig, WithShards,
+        },
+    };
+
+    #[tokio::test]
+    async fn hybrid_oprf() {
+        let world: TestWorld<WithShards<SHARDS, RoundRobinInputDistribution>> =
+            TestWorld::with_shards(TestWorldConfig::default());
+
+        let contexts = world.contexts();
+
+        const SHARDS: usize = 2;
+
+        let records = vec![
+            TestHybridRecord::TestImpression {
+                match_key: 12345,
+                breakdown_key: 2,
+            },
+            TestHybridRecord::TestImpression {
+                match_key: 68362,
+                breakdown_key: 1,
+            },
+            TestHybridRecord::TestConversion {
+                match_key: 12345,
+                value: 5,
+            },
+            TestHybridRecord::TestConversion {
+                match_key: 68362,
+                value: 2,
+            },
+            TestHybridRecord::TestImpression {
+                match_key: 68362,
+                breakdown_key: 1,
+            },
+            TestHybridRecord::TestConversion {
+                match_key: 68362,
+                value: 7,
+            },
+        ];
+
+        let shares: [Vec<HybridReport<BA8, BA3>>; 3] = records.iter().cloned().share();
+
+        let indistinguishable_reports: [Vec<IndistinguishableHybridReport<BA8, BA3>>; 3] = shares
+            .iter()
+            .map(|s| s.into_iter().map(|r| r.clone().into()).collect::<Vec<_>>())
+            .collect::<Vec<_>>()
+            .try_into()
+            .expect("Expected exactly 3 elements");
+
+        let chunked_reports: [Vec<Vec<IndistinguishableHybridReport<BA8, BA3>>>; 3] =
+            indistinguishable_reports
+                .iter()
+                .map(|vec| {
+                    let mid = vec.len() / 2;
+                    vec.chunks(mid)
+                        .map(|chunk| chunk.to_vec())
+                        .collect::<Vec<_>>()
+                })
+                .collect::<Vec<_>>()
+                .try_into()
+                .expect("Expected exactly 3 elements");
+
+        // #[allow(clippy::large_futures)]
+        // let results = flatten3v(chunked_reports.into_iter().zip(contexts).map(
+        //     |(reports_by_helper, helper_ctxs)| {
+        //         reports_by_helper
+        //             .into_iter()
+        //             .zip(helper_ctxs)
+        //             .map(|(reports, ctx)| compute_prf_for_inputs(ctx, &reports))
```

Comment:
this gives me a compile error:
```
error[E0515]: cannot return value referencing local variable `reports`
   --> ipa-core/src/protocol/hybrid/oprf.rs:247:43
    |
247 |                     .map(|(reports, ctx)| compute_prf_for_inputs(ctx, &reports))
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^--------^
    |                                           |                           |
    |                                           |                           `reports` is borrowed here
    |                                           returns a value referencing data owned by the current function
 ```

Reply:
you may need `async move { compute_prf_for_inputs(ctx, &reports).await }`  here - not 100% sure if it is going to help though

Reply:
i think that was it. fingers crossed